middle_end/flambda/types/env/typing_env.rec.ml

```diff
@@ -1084,8 +1095,8 @@ and add_equation t name ty =
 
      Note also that [p] and [x] may have different name modes! *)
   let ty, t =
-    let [@inline always] name eqn_name ~coercion:_ =
-      (* CR lmaurer: Coercion dropped! *)
+    let [@inline always] name eqn_name ~coercion =
+      assert (Coercion.is_id coercion); (* true by definition *)
```

Comment:
If this is true by definition do we really need to take [coercion] as an argument?

Reply:
Yes, only because `name` is going to be the argument to `Simple.pattern_match`.